src/appengine/libs/issue_management/oss_fuzz_github.py

```diff
@@ -0,0 +1,195 @@
+  return ISSUE_ClOSE_COMMENT_TEXT.format(
+      bug_information=testcase.bug_information)
+
+
+def _get_access_token():
+  """Get access to GitHub with the oss-fuzz personal access token"""
+  token = db_config.get_value('oss_fuzz_robot_github_personal_access_token')
+  if not token:
+    logs.log_error('Unable to get oss-fuzz-robot personal access token.')
+    return None
+  return github.Github(token)
+
+
+def _filing_enabled(testcase):
+  """Check if the project YAML file requires to file a GitHub issue."""
+  require_issue = data_handler.get_value_from_job_definition(
+      testcase.job_type, 'FILE_GITHUB_ISSUE', default='False')
+  return utils.string_is_true(require_issue)
+
+
+def _get_repo(testcase, access):
+  """Get the GitHub repository to file the issue"""
+  repo_url = data_handler.get_value_from_job_definition(testcase.job_type,
+                                                        'MAIN_REPO', '')
+  if not repo_url:
+    logs.log_error('Unable to fetch the MAIN_REPO URL from job definition.')
+    return None
+  repo_name = repo_url.removeprefix(GITHUB_PREFIX)
+
+  try:
+    target_repo = access.get_repo(repo_name)
+  except github.UnknownObjectException:
+    logs.log_error(f'Unable to locate GitHub repository '
+                   f'named {repo_name} from URL: {repo_url}.')
+    target_repo = None
+  return target_repo
+
+
+def _find_existing_issue(repo, issue_title):
+  """Checking if there is an existing open issue under the same name"""
+  for issue in repo.get_issues():
+    if issue.title == issue_title:
+      logs.log(f'Issue ({issue_title}) already exists in Repo ({repo.id}).')
+      return issue
+  return None
+
+
+def _post_issue(repo, testcase):
+  """Post the issue to the Github repo of the project."""
+  issue_title = get_issue_title(testcase)
+  issue_body = get_issue_body(testcase)
+  return _find_existing_issue(repo, issue_title) \
+      or repo.create_issue(title=issue_title, body=issue_body)
+
+
+def _update_testcase_properties(testcase, repo, issue):
+  """Update the GitHub-related properties in the FiledBug entity."""
+  testcase.repo_id = repo.id
+  testcase.issue_num = issue.number
+
+
+def file_issue(testcase):
+  """File an issue to the GitHub repo of the project"""
+  if not _filing_enabled(testcase):
+    return
+  if testcase.repo_id and testcase.issue_num:
+    logs.log('Issue already filed under'
+             f'issue number {testcase.issue_num} in Repo {testcase.repo_id}.')
+    return
+  access_token = _get_access_token()
+  if not access_token:
+    raise RuntimeError('Unable to access GitHub account and file the issue.')
+  repo = _get_repo(testcase, access_token)
+  if not repo:
+    logs.log_error('Unable to locate GitHub repository and file the issue.')
+    return
+  issue = _post_issue(repo, testcase)
+  _update_testcase_properties(testcase, repo, issue)
+
+
+def _issue_recorded(testcase):
+  """Verify the issue has been filed."""
+  return testcase.github_repo_id \
```

Comment:
I think we usually prefer to use paren to split up lines rather than backslash.
I think this is easier to read:
```
return (test.case.github_repo_id and
  testcase.github_issue_num)
```

Reply:
Thanks!
It turns out they are short enough to fit into one line : )
But I will use paren next time!